### PR TITLE
Bring back hive support for Python 3.11

### PR DESCRIPTION
**Pull request by potiuk**
---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

**Comment by potiuk** [comment]
ah @pankajastro I saw yo ualso looked at this one :)

**Comment by mdeshmu** [comment]
It should be pyhive[hive_pure_sasl]>=0.7.1.dev0

**Comment by potiuk** [comment]
> It should be pyhive[hive_pure_sasl]>=0.7.1.dev0

Yeah. Seems like we have to combine the two :)

**Comment by pankajastro** [comment]
> ah @pankajastro I saw yo ualso looked at this one :)

yeah, actually because apache-hive provider (sasl )only.

**Comment by potiuk** [comment]
ok. @mdeshmu - I updated the PR to the state as it **should** be merged when 0.7.1 final release is made. Let us know when  the release happen and we will merge it then.

File: airflow/providers/apache/hive/CHANGELOG.rst [hunk 1]
```diff
@@ -27,6 +27,15 @@
 Changelog
 ---------
 
+6.1.3
+.....
+
+The provider now uses pure-sasl, pure-python implementation of SASL,
```
Comment (uranusjr) [inline airflow/providers/apache/hive/CHANGELOG.rst]:
```suggestion
The provider now uses pure-sasl, a pure-Python implementation of SASL,
```

File: airflow/providers/apache/hive/provider.yaml [hunk 2]
```diff
@@ -58,18 +59,9 @@ dependencies:
   - apache-airflow-providers-common-sql>=1.3.1
   - hmsclient>=0.1.0
   - pandas>=0.17.1
-  - pyhive[hive]>=0.6.0
-  # in case of Python 3.9 sasl library needs to be installed with version higher or equal than
-  # 0.3.1 because only that version supports Python 3.9. For other Python version pyhive[hive] pulls
-  # the sasl library anyway (and there sasl library version is not relevant)
-  - sasl>=0.3.1; python_version>="3.9"
+  - pyhive[hive_pure_sasl]>=0.7.1
```
Comment (pankajkoti) [inline airflow/providers/apache/hive/provider.yaml:62]:
```suggestion
  - pyhive[hive_pure_sasl]>=0.7.1.dev0
```

**Comment by mdeshmu** [comment]
PyHive 0.7.0 is released which includes Python 3.11 and SQLAlchemy 2.0 support.

File: airflow/providers/apache/hive/provider.yaml [hunk 3]
```diff
@@ -60,18 +60,9 @@ dependencies:
...
```
Comment (potiuk) [inline airflow/providers/apache/hive/provider.yaml]:
```suggestion
  - pyhive[hive_pure_sasl]>=0.7.0
```

File: generated/provider_dependencies.json [hunk 4]
```diff
@@ -119,8 +119,7 @@
       "apache-airflow>=2.4.0",
       "hmsclient>=0.1.0",
       "pandas>=0.17.1",
-      "pyhive[hive]>=0.6.0",
-      "sasl>=0.3.1; python_version>=\"3.9\"",
+      "pyhive[hive_pure_sasl]>=0.7.1",
```
Comment (potiuk) [inline generated/provider_dependencies.json]:
```suggestion
      "pyhive[hive_pure_sasl]>=0.7.0",
```

**Comment by potiuk** [comment]
Seems that `pyhive 0.7.0 has been released and we can bring back Hive support for Python 3.11 finally

**Review by uranusjr — APPROVED** [review]

**Review by eladkal — APPROVED** [review]
LGTM
No need for 2.7.1 milestone it will be part of provider release cycle

**Comment by potiuk** [comment]
We will actually need it in 2.7 branch/when we prepare 2.7.1 release. 

We need to sync provider dependencies when we add a new provider (and this will need to be cherry-picked). The thing is that `provider_dependencies.json` is generated from `provider.yaml` files and it is used to generate `airflow's` extras. This is not needed for `airflow' package (because in the generated package, the dependencies are replaced with "apache-airflow-providers-apache-hive" dependency), but it matters for constraints generation. If we do not have "apache.hive" in 2-7 branch for Python 3.11 then Airflow 2.7.1 will not have constraints generated by hive even if the provider package with 3.11 support is released.

This is why I marked it with 2.7.1 - so that it gets cherry-picked there.

**Comment by potiuk** [comment]
Just flakes.

**Pull request merged**